app/models/sensor.rb

```diff
@@ -21,6 +21,51 @@ def self.create_from_params(params)
     return sensor
   end
 
+  def backfill_relay_times
+    conn = PG::Connection.open(ENV["RELAY_DB_URL"])
+    sql = ActiveRecord::Base::sanitize_sql_for_conditions(["select * from public.temperatures where cell_id = '%s'", self.name])
+    res = conn.exec(sql)
+    res.each do |relay_reading| 
+      ## TODO: make these all -0500 EST ?
+      # res.each{|r| puts "would have updated #{s.readings.find_by(created_at: r['hub_time']).created_at} to #{r['time']}" }
+      sensor_reading = self.readings.find_by(created_at: relay_reading['hub_time'])
+      if sensor_reading.nil?
+        puts "skipping sensor reading at #{relay_reading['hub_time']}, no match"
+      else
+        puts "Filling columns for reading at #{relay_reading['hub_time']}"
+        sensor_reading.update(
+          relay_received_at: relay_reading["time"], 
+          hs_received_at: relay_reading["relayed_time"], 
+          device_recorded_at: relay_reading["hub_time"]
+        )
+      end
+    end
+  end
+
+  def fix_improper_created_at_times
+    ## Fixing improper times on ESP32-S2 devices where the clock ran ahead
+    # created_at gets set to the time from the device
+    # If that's after relay_received_at, we know the time on the device was wrong
+    # in that case, use the time the relay received the data, as that was close to the reading time
+
+    self.readings.where("created_at > relay_received_at").each do |sensor_reading|
```

Comment:
how does this handle nulls?  should it be `relay_received_at is not null AND created_at > relay_received_at` or does it already ignore them?

Reply:
Good call out - I doublechecked and this only returns rows where relay_received_at is populated.